tests/out/glsl/atomicSimple.corge.Compute.glsl

```diff
@@ -0,0 +1,57 @@
+#version 310 es
+
+precision highp float;
+precision highp int;
+
+layout(local_size_x = 2, local_size_y = 1, local_size_z = 1) in;
+
+shared uint foo;
+
+shared uint baz[2];
+
+
+void main() {
+    if (gl_LocalInvocationID == uvec3(0u)) {
+        foo = 0u;
+        baz = uint[2](0u, 0u);
+    }
```

Comment:
question: I'm actually puzzled, but have a hypothesis as to why this code gets emitted, though there's no apparent equivalent in WGSL source. This appears to be automatically generated init., so variables have a well-known value. Does invocation `0` always run first? Is that right?

Reply:
Bringing back in context from sync discussion with @jimblandy: This is actually a _tie-breaker_ for the workgroup, where we elect one and only one thread/invocation in the thread group/workgroup to set the group's state properly.

I'm satisfied with what I've learned here. Thanks, Jim!

Reply:
Interestingly, Raph Levien [notes](https://stackoverflow.com/questions/39393560/glsl-memorybarriershared-usefulness#comment119242328_39395947) that in OpenGL 4.6 and GL ES 3.2, the `memoryBarrierShared` might not be necessary here. IDK what our deps are, so 🤷🏻‍♂️. Just interesting trivia, I guess?